code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -62,45 +62,43 @@
 			playsound(Y,'sound/effects/thunder.ogg', 90, 1)
 			spawn(10)
 				user.gib()
-		if(B.contents.len)
-			var/obj/item/O = pick(B.contents)
-			var/turf/location = get_turf(B)
+		var/obj/item/weapon/storage/internal/pocket/butt/theinv = B.inv
+		if(theinv.contents.len)
+			var/obj/item/O = pick(theinv.contents)
 			if(istype(O, /obj/item/weapon/lighter))
 				var/obj/item/weapon/lighter/G = O
-				if(G.lit && location)
-					new/obj/effect/hotspot(location)
+				if(G.lit && user.loc)
+					new/obj/effect/hotspot(user.loc)
 					playsound(user, 'sound/misc/fart.ogg', 50, 1, 5)
 			else if(istype(O, /obj/item/weapon/weldingtool))
 				var/obj/item/weapon/weldingtool/J = O
-				if(J.welding == 1 && location)
-					new/obj/effect/hotspot(location)
+				if(J.welding == 1 && user.loc)
+					new/obj/effect/hotspot(user.loc)
 					playsound(user, 'sound/misc/fart.ogg', 50, 1, 5)
 			else if(istype(O, /obj/item/weapon/bikehorn) || istype(O, /obj/item/weapon/bikehorn/rubberducky))
+				message = "<span class='clown'>farts.</span>"
 				playsound(user, 'sound/items/bikehorn.ogg', 50, 1, 5)

```

Comment:
We should definitely replace this with the bikehorn's special sound variable, so the sax makes it's random noise, etc.